http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/resources/HttpVersionClientResource.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @GET
+    @Path("http-asynchronous")
+    public Response httpAsynchronous() throws MalformedURLException {
+        return createAsynchronousResponseForClient(baseUri);
+    }
+
+    @GET
+    @Path("https-synchronous-for-client-with-key")
+    public Response httpsSynchronousForClientWithKey() {
+        Response response = httpsVersionClientWithConfigKey.getClientHttpVersion(NAME);
+        return checkHttpVersion(response);
+    }
+
+    @GET
+    @Path("http-synchronous-for-client-with-key")
+    public Response httpSynchronousForClientWithKey() {
+        Response response = httpVersionClientWithConfigKey.getClientHttpVersion(NAME);
+        return checkHttpVersion(response);
+    }
+
+    @GET
+    @Path("https-asynchronous-for-client-with-key")
+    public Response httpsAsynchronousForClientWithKey() {
+        Response response = httpsVersionClientWithConfigKey.getClientHttpVersionAsync(NAME)
+                .await().atMost(Duration.ofSeconds(TIMEOUT_SECONDS));
+
+        return checkHttpVersion(response);
+    }
+
+    @GET
+    @Path("http-asynchronous-for-client-with-key")
+    public Response httpAsynchronousForClientWithKey() {
+        Response response = httpVersionClientWithConfigKey.getClientHttpVersionAsync(NAME)
+                .await().atMost(Duration.ofSeconds(TIMEOUT_SECONDS));
+
+        return checkHttpVersion(response);
+    }
+
+    /**
+     * Create HttpVersionClient and make REST client synchronous request to `httpVersion/synchronous endpoint`
+     * Response from the client is checked if http 2 was used
+     *
+     * @param uri base url for client to use
+     * @return Response contain original response from endpoint or error message containing actual http version used
+     */
+    private Response createSynchronousResponseForClient(URI uri) throws MalformedURLException {
+        HttpVersionClient httpVersionClient = RestClientBuilder.newBuilder()
+                .baseUrl(uri.toURL())
+                .build(HttpVersionClient.class);
+
+        Response response = httpVersionClient.getClientHttpVersion(NAME);
+        return checkHttpVersion(response);
+    }
+
+    /**
+     * Create HttpVersionClient and make REST client asynchronous request to `httpVersion/asynchronous endpoint`
+     * Response from the client is checked if http 2 was used
+     *
+     * @param uri base url for client to use
+     * @return Response contain original response from endpoint or error message containing actual http version used
+     */
+    private Response createAsynchronousResponseForClient(URI uri) throws MalformedURLException {
+        HttpVersionClient httpVersionClient = RestClientBuilder.newBuilder()
+                .baseUrl(uri.toURL())
+                .build(HttpVersionClient.class);
+
+        Response response = httpVersionClient.getClientHttpVersionAsync(NAME).await()
+                .atMost(Duration.ofSeconds(TIMEOUT_SECONDS));
```

Comment:
I think blocking here is just laziness, there is no point to it. Asynchronous test should be processed in a reactive manner and return stuff asynchronously from resource as well. Please feel free to prove me wrong. Thanks